src/CLIFramework/Extension/DaemonExtension.php

```diff
@@ -0,0 +1,202 @@
+<?php
+namespace CLIFramework\Extension;
+
+use CLIFramework\ServiceContainer;
+use CLIFramework\Command;
+use CLIFramework\Logger;
+use CLIFramework\Exception\ExtensionException;
+use CLIFramework\IO\StreamWriter;
+
+class DaemonExtension extends ExtensionBase
+{
+    private $config;
+    private $logger;
+    private $isNoClose = false;
```

Comment:
Rename to `$detach = true`. if the daemon mode is enabled, we should detach by default.